--- /install/postscripts/otherpkgs	2024-09-03 10:47:58.920064505 +0530
+++ /install/postscripts/otherpkgs_updated	2024-09-03 10:45:43.521072346 +0530
@@ -598,6 +598,7 @@
 	       echo "enabled=1" >> $REPOFILE
 	       echo "gpgcheck=0" >> $REPOFILE
                echo "skip_if_unavailable=True" >> $REPOFILE
+	       echo "proxy=_none_" >> $REPOFILE
            fi
            i=$((i+1))
       done
@@ -663,6 +664,7 @@
                 echo "enabled=1" >> $REPOFILE
                 echo "gpgcheck=0" >> $REPOFILE
                 echo "skip_if_unavailable=True" >> $REPOFILE
+		echo "proxy=_none_" >> $REPOFILE
 
             elif [ $hasapt -eq 1 ] ; then 
                 REPOFILE="$repo_base/xCAT-otherpkgs${urlrepoindex}.list"
@@ -768,6 +770,7 @@
 	            echo "enabled=1" >> $REPOFILE
 	            echo "gpgcheck=0" >> $REPOFILE
                     echo "skip_if_unavailable=True" >> $REPOFILE
+                    echo "proxy=_none_" >> $REPOFILE
                 if [ $hasyum -eq 1 ]; then
                     yum clean all
                 fi
