--- debian.pm	2024-05-22 09:13:24.667052041 +0000
+++ debian_hwe.pm	2024-05-22 09:21:24.956009922 +0000
@@ -850,6 +850,11 @@
                             and -r "$pkgdir/install/netboot/initrd.gz"
                             and $initrdpath = "$pkgdir/install/netboot/initrd.gz"
                         ) or
+                        (-r "$pkgdir/casper/hwe-vmlinuz"
+                            and $kernpath = "$pkgdir/casper/hwe-vmlinuz"
+                            and -r "$pkgdir/casper/hwe-initrd"
+                            and $initrdpath = "$pkgdir/casper/hwe-initrd"
+                        ) or
                         (-r "$pkgdir/casper/vmlinuz"
                             and $kernpath = "$pkgdir/casper/vmlinuz"
                             and -r "$pkgdir/casper/initrd"
