--- configib	2024-08-05 15:26:56.369733272 +0000
+++ updated_configib	2024-08-05 15:28:20.045752732 +0000
@@ -21,6 +21,7 @@
 if [ "$(uname -s|tr 'A-Z' 'a-z')" = "linux" ];then
    str_dir_name=`dirname $0`
    . $str_dir_name/xcatlib.sh
+   . $str_dir_name/nicutils.sh
 fi
 
 if [ -n "$LOGLABEL" ]; then
@@ -610,6 +611,36 @@
                 fi # end not the first ip address
             elif [ $OS_name == 'ubuntu' ]
             then
+                netplan info
+                netplan_active=$?
+                if [ $netplan_active -eq 0 ]; then
+                    #configuring the ib nic when netplan is active
+                    calculate_netmask_bits $netmask
+                    netmask_bits=$?
+
+                    str_conf_file=/etc/netplan/${nic}.yaml
+                    rm -rf /etc/netplan/${nic}.yaml
+                    echo "network:"  > $str_conf_file
+                    echo "  version: 2" >> $str_conf_file
+                    echo "  ethernets:" >> $str_conf_file
+                    echo "    ${nic}:" >> $str_conf_file
+                    echo "      addresses:" >> $str_conf_file
+                    echo "      - ${nicip}/${netmask_bits}"  >> $str_conf_file
+                    echo "      dhcp4: false"  >> $str_conf_file
+
+                    if [ -n "${mtu}" ];then
+                        echo "      mtu: ${mtu}"  >> $str_conf_file
+                    fi
+                    if [ -n "${gateway}" ];then
+                        echo "      routes:" >> $str_conf_file
+                        echo "      - to: default" >> $str_conf_file
+                        echo "        via: ${gateway}" >> $str_conf_file
+                    fi
+
+                    chmod 600 $str_conf_file
+
+                    netplan apply
+                else
                 # First ip address
                 if [ $ipindex -eq 1 ]; then
                     # ipv6
@@ -657,7 +688,7 @@
                         #    echo "gateway $gateway" >> /etc/network/interfaces
                         #fi
                     fi
-				fi
+		fi
 				#add extra params
 				i=0
 				while [ $i -lt ${#array_extra_param_names[@]} ]
@@ -673,7 +704,7 @@
                                         fi
 					i=$((i+1))
 				done
-
+		fi
            else
                 echo "Unsupported operating system"
                 logger -p local4.err -t $log_label "Unsupported operating system"
@@ -770,7 +801,10 @@
                     sleep 2
                     ifup   $tmp > /dev/null 2>&1
                 done
-            else 
+            elif [ $OS_name == 'ubuntu' ]
+            then
+		netplan apply
+            else
                 if [ $nmcli_used -eq 1 ]; then
                     nmcli con reload $dir/ifcfg-$nic
                     nmcli con up $nic 2>&1
