--- confignetwork       2024-12-18 17:56:59.297612202 +0000
+++ updated_confignetwork       2024-12-18 18:56:55.449911749 +0000
@@ -46,7 +46,13 @@
     is_sles=1
     nwdir="/etc/sysconfig/network"
 elif [ -f "/etc/debian_version" ];then
-    nwdir="/etc/network/interfaces.d"
+    netplan info
+    netplan_active=$?
+    if [ $netplan_active -eq 0 ]; then
+        nwdir="/etc/netplan"
+    else
+        nwdir="/etc/network/interfaces.d"
+    fi
     is_debian=1
 else
     log_error "Only supports Linux"
@@ -64,18 +70,31 @@
     if [ $is_redhat -eq 1 ] || [ $is_sles -eq 1 ]; then
          cfg_file="$nwdir/ifcfg-${cfg_dev}"
     elif [ $is_debian -eq 1 ]; then
-         cfg_file="$nwdir/${cfg_dev}"
+        netplan info
+        netplan_active=$?
+        if [ $netplan_active -eq 0 ]; then
+            cfg_file="$nwdir/${cfg_dev}.yaml"
+        else
+            cfg_file="$nwdir/${cfg_dev}"
+        fi
     fi
     if [ "$networkmanager_active" != "0" ]; then
         $ip address show dev ${cfg_dev}| $sed -e 's/^/[Ethernet] >> /g' | log_lines info
     else
-        if [ -f $cfg_file ]; then
-            echo "['${cfg_file}']" >&2
-            cat ${cfg_file}| $sed -e 's/^/ >> /g' | log_lines info
-        else
-            log_error "Can not find $cfg_file."
-            errorcode=1
-        fi
+        netplan info
+         netplan_active=$?
+         if [ $netplan_active -eq 0 ]; then
+           echo "netplan active"
+        else
+           if [ -f $cfg_file ]; then
+             echo "['${cfg_file}']" >&2
+             cat ${cfg_file}| $sed -e 's/^/ >> /g' | log_lines info
+           else
+             log_error "Can not find $cfg_file."
+             errorcode=1
+           fi
+        fi
+
     fi
 }

@@ -361,7 +380,7 @@
     fi

     if [ -n "$nics_list" ]; then
-      if [ $is_redhat -eq 1 ]; then
+      if [ $is_redhat -eq 1 ] || [ $is_debian -eq 1 ]; then
         num=1
         max=`echo "$nics_list"|wc -l`
         ((max+=1))
@@ -547,6 +566,7 @@
                 log_info "configure $nic_dev"
                 log_info "call: NMCLI_USED=$networkmanager_active configeth $nic_dev $ipaddrs $xcatnet"
                 NMCLI_USED=$networkmanager_active configeth $nic_dev $ipaddrs $xcatnet
+               ip address show dev $nic_dev
                 if [ $? -ne 0 ]; then
                     errorcode=1
                 fi
@@ -591,10 +611,16 @@
                 vlanname=`echo "$nic_dev" | $sed -e 's/^\(.*\)vla\?n\?\([0-9]\+\)$/\1/'`
             fi
             ipaddrs=$(find_nic_ips $nic_dev)
-            if [ "$networkmanager_active" != "1" ]; then
-                create_vlan_interface ifname=$vlanname vlanid=$vlanid
+            netplan info
+            netplan_active=$?
+            if [ $netplan_active -eq 0 ]; then
+                create_vlan_interface_netplan ifname=$vlanname vlanid=$vlanid ipaddrs=$ipaddrs next_nic=$is_mid_device
             else
-                create_vlan_interface_nmcli ifname=$vlanname vlanid=$vlanid ipaddrs=$ipaddrs next_nic=$is_mid_device
+                if [ "$networkmanager_active" != "1" ]; then
+                    create_vlan_interface ifname=$vlanname vlanid=$vlanid
+                else
+                    create_vlan_interface_nmcli ifname=$vlanname vlanid=$vlanid ipaddrs=$ipaddrs next_nic=$is_mid_device
+                fi
             fi
             if [ $? -ne 0 ]; then
                 log_error "configure VLAN failed."
