--- ddns.pm	2024-10-21 06:58:33.663187950 +0000
+++ updated_ddns.pm	2024-10-21 13:38:23.494393637 +0000
@@ -1284,11 +1284,15 @@
             push @newnamed, "\t};\n";
             my $bind_version_cmd="/usr/sbin/named -v | cut -d' ' -f2 | grep -Eo '[0-9]+\.[0-9]+\.[0-9]+'";
             my @bind_version =xCAT::Utils->runcmd($bind_version_cmd, 0);
-            # Turn off DNSSEC if running with bind vers 9.16.6 or higher
-            if ((scalar @bind_version > 0) && (xCAT::Utils::CheckVersion($bind_version[0], "9.16.6") >= 0)) {
+	    # Turn off DNSSEC if running with bind version greater than or equal to 9.16.6 and less than 9.18.0
+            if ((scalar @bind_version > 0) && (xCAT::Utils::CheckVersion($bind_version[0], "9.16.6") >= 0) && (xCAT::Utils::CheckVersion($bind_version[0], "9.18.0") < 0)) {
                 push @newnamed, "\tdnssec-enable no;\n";
                 push @newnamed, "\tdnssec-validation no;\n";
             }
+            # Turn off DNSSEC validation if running with bind version greater than or equal to 9.18.0
+            if ((scalar @bind_version > 0) && (xCAT::Utils::CheckVersion($bind_version[0], "9.18.0") >= 0)) {
+                push @newnamed, "\tdnssec-validation no;\n";
+            }
         }
 
         if ($ctx->{forwardmode}){
@@ -1564,7 +1568,14 @@
 
                     # sometimes even the xcat_key is correct, but named still replies NOTAUTH, so retry
                     for (1 .. 3) {
-                        $update->sign_tsig("xcat_key", $ctx->{privkey});
+                        my $key_name = 'xcat_key';
+                        my $key      = $ctx->{privkey};
+                        my $tsig = Net::DNS::RR->new(
+                            name => $key_name,
+                            type => 'TSIG',
+                            key  => $key
+                        );
+                        $update->sign_tsig($tsig);
                         $numreqs = 300;
                         my $reply = $resolver->send($update);
                         if ($reply) {
@@ -1586,8 +1597,15 @@
             if ($numreqs != 300) { #either no entries at all to begin with or a perfect multiple of 300
                  # sometimes even the xcat_key is correct, but named still replies NOTAUTH, so retry
                 for (1 .. 3) {
-                    $update->sign_tsig("xcat_key", $ctx->{privkey});
-                    my $reply = $resolver->send($update);
+                    my $key_name = 'xcat_key';
+                    my $key      = $ctx->{privkey};
+                    my $tsig = Net::DNS::RR->new(
+                        name => $key_name,
+                        type => 'TSIG',
+                        key  => $key
+                    );
+                    $update->sign_tsig($tsig);
+		    my $reply = $resolver->send($update);
                     if ($reply) {
                         if ($reply->header->rcode eq 'NOTAUTH') {
                             next;
