--- ddns.pm	2024-08-29 03:48:16.270158152 +0000
+++ updated_ddns.pm	2024-08-29 03:48:42.450154921 +0000
@@ -1284,11 +1284,15 @@
             push @newnamed, "\t};\n";
             my $bind_version_cmd="/usr/sbin/named -v | cut -d' ' -f2 | grep -Eo '[0-9]+\.[0-9]+\.[0-9]+'";
             my @bind_version =xCAT::Utils->runcmd($bind_version_cmd, 0);
-            # Turn off DNSSEC if running with bind vers 9.16.6 or higher
-            if ((scalar @bind_version > 0) && (xCAT::Utils::CheckVersion($bind_version[0], "9.16.6") >= 0)) {
+	    # Turn off DNSSEC if running with bind version greater than or equal to 9.16.6 and less than 9.18.0
+            if ((scalar @bind_version > 0) && (xCAT::Utils::CheckVersion($bind_version[0], "9.16.6") >= 0) && (xCAT::Utils::CheckVersion($bind_version[0], "9.18.0") < 0)) {
                 push @newnamed, "\tdnssec-enable no;\n";
                 push @newnamed, "\tdnssec-validation no;\n";
             }
+            # Turn off DNSSEC validation if running with bind version greater than or equal to 9.18.0
+            if ((scalar @bind_version > 0) && (xCAT::Utils::CheckVersion($bind_version[0], "9.18.0") >= 0)) {
+                push @newnamed, "\tdnssec-validation no;\n";
+            }
         }
 
         if ($ctx->{forwardmode}){
