--- pgsqlsetup	2024-07-25 11:11:49.793271811 +0000
+++ updated_pgsqlsetup	2024-07-25 11:12:06.846577333 +0000
@@ -1060,7 +1060,7 @@
     #$pgsql->log_stdout(1);
     $pgsql->log_stdout($debug);
 
-    my $createuser = "CREATE USER xcatadm WITH PASSWORD \'$::adminpassword\';\r";
+    my $createuser = "CREATE USER xcatadm WITH SUPERUSER PASSWORD \'$::adminpassword\';\r";
     my $spawncmd;
     $spawncmd = $cmd;
     unless ($pgsql->spawn($spawncmd))
