--- doxcat	2023-09-11 09:55:53.739008109 +0530
+++ updatedoxcat	2023-09-11 09:55:36.110898984 +0530
@@ -423,12 +423,14 @@
 		if [ $IPMI_SUPPORT -ne 0 ]; then
                     # Set boot from network will cause OpenPOWER server wait at petitboot menu, so do nothing here
                     if uname -m | grep x86_64; then
+                        ipmitool raw 0x00 0x08 0x03 0x08
 		        ipmitool chassis bootdev pxe
                     fi
 		fi
 		reboot -f
 	elif [ "$dest" = "install" -o "$dest" = "netboot" ]; then
 		if [ $IPMI_SUPPORT -ne 0 ]; then
+                    ipmitool raw 0x00 0x08 0x03 0x08
 		    ipmitool chassis bootdev pxe
 		fi
 		logger -s -t $log_label -p local4.info "Reboot..."
