--- configeth	2023-09-07 21:21:38.309242669 +0530
+++ updateconfigeth	2023-09-07 21:23:26.500943581 +0530
@@ -793,6 +793,7 @@
             ip link set dev $str_inst_nic down
         fi
         if [ $networkmanager_active -eq 1 ]; then
+            nmcli con modify $con_name ipv4.dns "${NAMESERVERS}"
             nmcli con reload
             nmcli con up $con_name
 	    else
@@ -1213,6 +1214,7 @@
                 if_state=0
                 echo "bring up ip"
                 if [ $networkmanager_active -eq 1 ]; then
+                    nmcli con modify $con_name ipv4.dns "${NAMESERVERS}"
                     nmcli con reload
                     nmcli con up $con_name
                 else
